components/dashboard/src/teams/SSO.tsx

```diff
@@ -0,0 +1,349 @@
+        | { mode: "new" }
+        | { mode: "edit"; clientConfig: OIDCClientConfig }
+        | { mode: "delete"; clientConfig: OIDCClientConfig }
+        | undefined
+    >(undefined);
+
+    useEffect(() => {
+        reloadClientConfigs().catch(console.error);
+    }, []);
+
+    const reloadClientConfigs = async () => {
+        const clientConfigs = await oidcService
+            .listClientConfigs({ organizationId: props.organizationId })
+            .then((resp) => {
+                return resp.providers;
+            });
+        setClientConfigs(clientConfigs);
+    };
+
+    const loginWith = (id: string) => {
+        window.location.href = gitpodHostUrl.with({ pathname: `/iam/oidc/start`, search: `id=${id}` }).toString();
+    };
+
+    const configMenu = (clientConfig: OIDCClientConfig) => {
+        const result: ContextMenuEntry[] = [];
+        result.push({
+            title: "Login",
+            onClick: () => loginWith(clientConfig.id),
+            separator: true,
+        });
+        result.push({
+            title: "Edit",
+            onClick: () => setModal({ mode: "edit", clientConfig }),
+            separator: true,
+        });
+        result.push({
+            title: "Remove",
+            customFontStyle: "text-red-600 dark:text-red-400 hover:text-red-800 dark:hover:text-red-300",
+            onClick: () => setModal({ mode: "delete", clientConfig }),
+        });
+        return result;
+    };
+
+    return (
+        <>
+            {modal?.mode === "new" && (
+                <OIDCClientConfigModal
+                    mode={modal.mode}
+                    organizationId={props.organizationId}
+                    onClose={() => setModal(undefined)}
+                    onUpdate={reloadClientConfigs}
+                />
+            )}
+
+            {modal?.mode === "edit" && <></>}
+            {modal?.mode === "delete" && <></>}
+
+            <div className="flex items-start sm:justify-between mb-2">
+                <div>
+                    <h3>Single sign sign-on with OIDC</h3>
+                    <h2>Setup SSO for your organization.</h2>
+                </div>
+                {clientConfigs.length !== 0 ? (
+                    <div className="mt-3 flex mt-0">
+                        <button onClick={() => setModal({ mode: "new" })} className="ml-2">
+                            New OIDC Client
+                        </button>
+                    </div>
+                ) : null}
+            </div>
+
+            {clientConfigs.length === 0 && (
+                <div className="w-full flex h-80 mt-2 rounded-xl bg-gray-100 dark:bg-gray-900">
+                    <div className="m-auto text-center">
+                        <h3 className="self-center text-gray-500 dark:text-gray-400 mb-4">No OIDC Clients</h3>
+                        <div className="text-gray-500 mb-6">
+                            Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor
```

Comment:
@jldec and @gtsiolis, please help with wording on this page.

Reply:
#15995